usaspending_api/etl/tests/conftest_spark.py

```diff
@@ -160,7 +171,44 @@ def spark() -> SparkSession:
 
 
 @fixture
-def delta_lake_unittest_schema(spark: SparkSession):
+def hive_unittest_metastore_db(spark: SparkSession):
+    """A fixture that WIPES all of the schemas (aka databases) and tables in each schema from the hive metastore_db
+    at the end of each test run, so that the metastore is fresh.
+
+    NOTE: This relies on setup in the session-scoped ``spark`` fixture:
+      - That fixture must enableHiveSupport() when creating the SparkSession
+      - That fixture needs to set the filesystem location of the hive metastore_db (Derby DB) folder in a tmp dir
+        - (so that it doesn't interfere or leave cruft behind)
+      - That fixture needs to set the Spark SQL Warehouse dir in a tmp dir
+        - (so that it doesn't interfere or leave cruft behind)
+
+    WARNING: If the spark test fixture is not setup to initialize the hive metastore_db in this way for the
+    SparkSession used by tests, then this fixture may inadvertently wipe all hive schemas and tables in you dev env
+    """
+    metastore_db_path = None
+    metastore_db_url = spark.conf.get("spark.hadoop.javax.jdo.option.ConnectionURL")
+    if metastore_db_url:
+        metastore_db_path = metastore_db_url.split("=")[1].split(";")[0]
+
+    yield metastore_db_path
+
+    schemas_in_metastore = [s[0] for s in spark.sql("SHOW SCHEMAS").collect()]
```

Comment:
Interesting. I didn't realize that `SCHEMA` was the preferred term in Databricks. I thought it was `DATABASE`.
https://docs.databricks.com/spark/latest/spark-sql/language-manual/sql-ref-syntax-aux-show-databases.html

Reply:
Yeah, I'm getting used to that too. Unfortunately, some of the internals (like outputting the row from a query for "SHOW TABLES" still has columns named `"database"` instead of `"schema"`.